custom_components/foxess_modbus/entities/modbus_select.py

```diff
@@ -57,6 +60,10 @@ def __init__(
     @property
     def current_option(self) -> str | None:
         value = self._controller.read(self.entity_description.address)
+        if value is None:
+            return None
+        if not self._validate(self.entity_description.validate, value):
+            return None
         return self.entity_description.options_map.get(value)
```

Comment:
This line is effectively checking whether the value is one that's valid, as it returns `None` if the key isn't found in `options_map`. Maybe we should add some logging around this?

Reply:
I went for the belt & braces approach and ensured consistency of validation across the code base - we could end up doing something different with the options map in the future (which doesn't return None by default) and/or need additional validation rules for other selects?

Reply:
`options_map.get` does return None if the key isn't found -- that's why I used `.get`. Agreed we might want additional validation rules in the future (although I can't think what they might be at the moment!)

I just dislike having duplicate knowledge spread around, as someone needs to manually keep them in sync. We already need knowledge of which values are valid so that we can map them onto strings, IMO duplicating that knowledge into a `Range` is just increasing the maintenance burden.